### DISCUSSION TITLE
Issue with calling the VanillaCNN nn

### DISCUSSION THREAD

**Discussion by asnokhai** [tudelft/AE4353-Y24#28] [Ex_2]
I have been struggling for a long time with this issue, and I am pretty stuck.

When I set 'model/backbone' to 'resnet', everything functions fine.

![image](https://github.com/user-attachments/assets/e4f6a4ad-d62b-4d8d-a45b-310089f171c4)

However, when I set it to 'vanilla', I get an AttributeError:

![image](https://github.com/user-attachments/assets/8a729a18-1994-449c-bc84-dc7379acf9cf)

I initially thought this was due to my implementation of 'VanillaCNN'. However, when I swap the names of both neural networks, the error remains unchanged:

![image](https://github.com/user-attachments/assets/b96ce7cb-3c40-451d-bdcc-fec4d2e690a0)

Here, effectively the 'VanillaCNN' carries the resnet implementation and vice versa. If I now run the code, it works with 'resnet' in the .yaml config, but I get the exact same error as before with 'vanilla'.

Before each run attempt, I cleared all outputs and restarted the python kernel.

How do I fix this?

Thanks in advance!

**Comment by yilun-wu** [comment]
As the error message suggests, you attempted to `squeeze` a `None` object instead of a tensor. From the error trace, it seems like it is the `output_angle` which is `None` that get squeezed but failed. Can you check whether that is is the case? And if so, further debug why `output_angle` is `None` instead of a pytorch tensor.

**Reply by asnokhai** [reply]
It is because model(x) returns 'None', but I can't seem to figure out why
